packages/starknet-snap/src/utils/starknetUtils.ts

```diff
@@ -747,18 +802,45 @@ export const getCorrectContractAddress = async (network: Network, publicKey: str
       );
       address = contractAddressLegacy;
     } catch (e) {
-      if (!e.message.includes('Contract not found')) {
+      if (e.message.includes('network error for getSigner')) {
```

Comment:
what is the reason to change this error, doesnt it throw Contract not found?

as we wanna to throw any error if it is not contract not found (means acc deployed, but something bad happend)